lib/board.rb

```diff
@@ -0,0 +1,65 @@
+class Board
+  attr_accessor :cells
+
+  def initialize()
+    reset!
+  end
+
+  def reset!
+    @cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts '-----------'
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts '-----------'
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+    puts "\n"
+  end
+
+  def input_index(input) # this one line is all over the place!
+    input.to_i - 1
+  end
+
+  # position method
+  def position(input)
+    @cells[input_index(input)]
+  end
+
+  # define full here - every element on the board contains "X" or "O"
+  def full?
+    @cells.all? do |mark|
+      mark == 'X' || mark == 'O'
+    end
+  end
+
+  # define turn_count here
+  def turn_count
+    counter = 0
```

Comment:
Consider using the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |occupied_spot| occupied_spot != " " }
end
```